lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1393,4 +1399,21 @@ public function __clone()
             $this->params[$name] = clone $param;
         }
     }
+
+    /**
+     * Filters out empty values from the given argument list.
+     * Fixes a BC break introduces in 2.11: https://github.com/doctrine/dbal/issues/4282
+     *
+     * @param array<mixed> $args
+     *
+     * @return array<mixed>
+     *
+     * @todo remove for 3.0
+     */
+    private static function filterEmptyArgs(array $args): array
+    {
+        return array_values(array_filter($args, static function ($value) {
+            return ! empty($value);
```

Comment:
Isn't it the default behavior of `array_filter` without a callback?

Reply:
Indeed! Updated. Should we keep the `filterEmptyArgs()` anyway, even if it's very simple, just as documentation?

Reply:
If it ends up being just `array_filter()`, it could be used directly w/o being wrapped in a documenting method. But I don't have a strong opinion.

Reply:
Okay, I've inlined it with a link to the issue and a TODO. See if you're happy with that!

Reply:
Looks good.